helix-term/src/commands.rs

```diff
@@ -1000,31 +1000,70 @@ fn goto_file_vsplit(cx: &mut Context) {
     goto_file_impl(cx, Action::VerticalSplit);
 }
 
+fn resolve_selected_path(
+    selection_text: String,
+    doc_path_parent: Option<&Path>,
+) -> Option<PathBuf> {
+    let trimmed = selection_text.trim();
+    if trimmed.is_empty() {
+        return None;
+    }
+
+    let selected_path = PathBuf::from(trimmed);
+    match (selected_path.is_relative(), doc_path_parent) {
+        (true, Some(parent_path)) => {
+            let mut result = PathBuf::new();
+            result.push(parent_path);
+            result.push(selected_path);
+            Some(result)
+        }
+        _ => Some(selected_path),
+    }
+}
+
 fn goto_file_impl(cx: &mut Context, action: Action) {
     let (view, doc) = current_ref!(cx.editor);
     let text = doc.text();
     let selections = doc.selection(view.id);
-    let mut paths: Vec<_> = selections
-        .iter()
-        .map(|r| text.slice(r.from()..r.to()).to_string())
-        .collect();
     let primary = selections.primary();
+    let doc_path_parent = doc.path().and_then(|path| path.parent());
+
     if selections.len() == 1 && primary.to() - primary.from() == 1 {
-        let current_word = movement::move_next_long_word_start(
+        let current_word = textobject::textobject_word(
             text.slice(..),
-            movement::move_prev_long_word_start(text.slice(..), primary, 1),
+            primary,
+            textobject::TextObject::Inside,
             1,
+            true,
         );
-        paths.clear();
-        paths.push(
-            text.slice(current_word.from()..current_word.to())
-                .to_string(),
-        );
-    }
-    for sel in paths {
-        let p = sel.trim();
-        if !p.is_empty() {
-            if let Err(e) = cx.editor.open(PathBuf::from(p), action) {
+        let selection_text = text
+            .slice(current_word.from()..current_word.to())
+            .to_string();
+        if let Some(path) = resolve_selected_path(selection_text, doc_path_parent) {
+            match path.metadata() {
+                Ok(metadata) => {
+                    if metadata.is_dir() {
+                        let picker = ui::file_picker(path, &cx.editor.config());
+                        cx.push_layer(Box::new(overlayed(picker)));
+                    } else if let Err(e) = cx.editor.open(path, action) {
+                        cx.editor.set_error(format!("Open file failed: {:?}", e));
+                    }
```

Comment:
Maybe allow opening the file by creating it even though it doesn't exist? IIRC in vim I have the need to open file even though they didn't exist, not sure how good is it to have it here.

Reply:
At least neovim also just reports "Can't find file in path" and doesn't try to create a file.

Thought experiment: If you pressed `gf` on a random string (maybe accidentally) and it would just create a file with this
name and place you in a buffer for it, I suspect, that could get kinda annoying.

Reply:
> Thought experiment: If you pressed gf on a random string (maybe accidentally) and it would just create a file with this
name and place you in a buffer for it, I suspect, that could get kinda annoying.

It sounds more like a feature than a bug for me.

Reply:
That's a big thing for us to decide I think.

Should we open a github discussion about it and see what others think?

It doesn't have to be settled with this PR.

Reply:
Okay, should be fine, not in this pull request.